generic_server.cc

```diff
@@ -224,6 +225,8 @@ future<> connection::process()
 
 void connection::on_connection_ready()
 {
+    _conns_cpu_concurrency.stopped = true;
+    _conns_cpu_concurrency.units.return_all();
```

Comment:
What happens if the connection does not reach the ready stage?

1. If it breaks 
2. If it just hangs during negotiation

Reply:
2 isn't a problem, since it can only hang while at the client side.

Reply:
then connection class is destroyed and _conns_cpu_concurrency.units returned